src/custom-lists/background/storage.ts

```diff
@@ -0,0 +1,191 @@
+                { field: 'name', unique: true },
+                { field: 'isDeletable' },
+                { field: 'isNestable' },
+                { field: 'createdAt' },
+            ],
+        })
+
+        this.storageManager.registerCollection(PAGE_LIST_ENTRY, {
+            // different version for adding a new table.
+            version: new Date(2018, 6, 12),
+            fields: {
+                listId: { type: 'string' },
+                pageUrl: { type: 'string' },
+                fullUrl: { type: 'string' },
+                createdAt: { type: 'datetime' },
+            },
+            indices: [
+                { field: ['listId', 'pageUrl'], pk: true },
+                { field: 'listId' },
+                { field: 'pageUrl' },
+            ],
+        })
+    }
+
+    // Return all the list in the DB
+    //  TODO: Use pagination if required and decide correct types
+    async fetchAllList({ query = {}, opts = {} }: { query?: any; opts?: any }) {
+        const x = await this.storageManager.findAll(COLLECTION_NAME, query, opts)
+        return this.changeListsbeforeSending(x, [])
+    }
+
+    // Takes the list as they come from Db and does some pre-processing before sending.
+    async changeListsbeforeSending(lists: object[], pageEnteries: object[]) {
+        const mappedLists = lists.map((list: ListObject) => {
+            const page = pageEnteries.find(({ listId }: PageObject) => listId === list.id)
+            delete list["_name_terms"]
+            return {
+                ...list,
+                // Set an empty pages array for manupulating in redux store.
+                pages: [],
+                active: Boolean(page),
+            }
+        })
+        return mappedLists
+    }
+
+    async fetchListById(id: Number) {
+        const list = await this.storageManager.findObject(COLLECTION_NAME, { id })
+        const pages = await this.storageManager.findAll(PAGE_LIST_ENTRY, { listId: list.id })
+        delete list["_name_terms"]
+        return {
+            ...list,
+            pages: pages.map((page: PageObject) => page.fullUrl),
+        }
+    }
+
+    // Return all the pages associated with a list.
+    async fetchListPages(listId: Number) {
+        return await this.storageManager.findAll(PAGE_LIST_ENTRY, { listId })
+    }
+
+    // Returns all the lists containing a certain page.
+    async getListAssocPage({ url }: { url: string }) {
+        const pages = await this.storageManager.findAll(PAGE_LIST_ENTRY, {
+            pageUrl: url,
+        })
+        const listIds = pages.map(({ listId }: PageObject) => listId)
+        const lists = await this.fetchAllList({
+            query: {
+                id: { $in: listIds }
+            }
+        })
+        return this.changeListsbeforeSending(lists, pages)
+    }
+
+    // Function to insert into the DB
+    async insertCustomList({ name, isDeletable = 1, isNestable = 1 }: {
+        name: string, isDeletable: 0 | 1, isNestable: 0 | 1
```

Comment:
Is there a reason to be storing these as `0 | 1` here over using standard boolean values?

Reply:
I did some research and could not figure out a way to store and index boolean in indexedDB, this was the other way around.

Reply:
TIL! Indeed, it seems like booleans are pretty useless in IDB and Dexie https://github.com/dfahlander/Dexie.js/issues/70

This would actually be something we should add field pre-processing support for in the StorageManager, so we don't have to write JS code that uses 1s and 0s to represent boolean values. I see you added the `binary` field type. I would argue a better field type name would be `bool`, just to avoid confusion with binary data types (like [`bytea` in postgres](https://www.postgresql.org/docs/9.1/static/datatype-binary.html)) which we may want to add support for. 

Anyway, I'll add this to my StorageManager TODOs, to auto-convert `bool` fields to 1s and 0s, and try to get done soon and then we can come back to this.